edb/edgeql/compiler/clauses.py

```diff
@@ -48,7 +49,7 @@ def compile_where_clause(
     with ctx.newscope(fenced=True) as subctx:
         subctx.path_scope.unnest_fence = True
         ir_expr = dispatch.compile(where, ctx=subctx)
-        bool_t = ctx.env.get_track_schema_type('std::bool')
+        bool_t = ctx.env.get_track_schema_type(sn.QualifiedName('std', 'bool'))
```

Comment:
Would be cool to have `sn.qname('std::bool')` with an LRU cache inside. That would save some keystrokes and maybe even make things a bit faster.

Reply:
:+1:

Reply:
I dunno.

`sn.QualName('std', 'bool')`
`sn.qname('std::bool')` 

Doesn't seem like it's worth an extra wrapper.  Even `lru_cache` has a cost.

Reply:
imo the latter is easier to type; python's lru_cache is in C and so caching by a str is extremely fast.

Reply:
It's also memory cost. Compilers aren't lightweight already. So I'm with @elprans here.

Reply:
Aside from performance considerations, I specifically didn't want to keep it easy to treat names as unparsed strings.  A handy casting shortcut creates too much incentive to keep names as strings.  Also, there aren't _that_ many places where the name of the object is a hardcoded string.